### ISSUE TITLE
Image block: "Unexpected error" when transforming to Carousel CoBlocks 

### ISSUE THREAD

**Issue by galakhyati** [Automattic/wp-calypso#56559]
### Quick summary

When switching from Image block to (CoBlocks) Carousel block they are thrown out of the editor.

Video

https://user-images.githubusercontent.com/7382505/134757029-46d64f32-b5e2-4b03-86fa-e666a4a4281b.mp4

:

### Steps to reproduce

1. Create a new page or edit an existing page
2. Add Image block, select the image.
3. Switch the block type and select Carousel.
4. You'll be thrown out of the editor and shows the error `The editor has encountered an unexpected error.`.

### What you expected to happen

The image should switch from an Image block to a carousel block with only one image. 

### What actually happened

An unexpected error was shown on the editor. 

Even after exiting the editor and trying to make any other edits on the page I'm shown the error again.

### Context

Customer report

### Simple, Atomic or both?

Atomic

Doesn't apply on Simple sites as the Carousel block isn't available.

### Other notes

There were errors in the console: 

![Markup on 2021-09-25 at 09:25:41](https://user-images.githubusercontent.com/7382505/134757042-76301e3d-20ef-4498-95e1-0f2e6de5b701.png)

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Disable the Gutenberg plugin.

#4324130-zen

**Comment by formosattic** [comment]
Thanks @galakhyati !

I was able to reproduce it.
I've updated the initial report with the workaround of disabling Gutenberg.

Also test on a self-hosted WordPress.org site, same behavior happens, the error only shows when Gutenberg is active.

Also reported in https://github.com/godaddy-wordpress/coblocks/issues/2042

**Comment by jordesign** [comment]
Relevant issue to check has been changed to https://github.com/godaddy-wordpress/coblocks/issues/2077 but is still open.

**Comment by msilbers** [comment]
This was reported in 32498798-hc by a customer. Although it was not the same action, all the errors listed above are exactly the same, and matches what's shown in the godaddy issue

The error occurs when trying to apply any filter (the color filters) to a Carousel block

**Comment by rw-ye** [comment]
28121350-hc
4435639-zd-woothemes

Same error message when switching from the galley to collage. Tested it on my own AT site as well.

**Comment by masperber** [comment]
33119373-hc Gets the same error when switching from an Image to a Carousel block.

I also reproduced the issue on my test site.

**Comment by masperber** [comment]
33555715-hc encountered this issue

**Comment by jartes** [comment]
I also had another report here: 33739949-hc. Follow-up: 4700710-zd-woothemes

But, in this case, it's happening when after "hovering" the option and the mouse "leaves" the Carousel option:

https://user-images.githubusercontent.com/1046943/150180175-c8c0706e-ac53-4365-b6f0-6440e9074a96.mp4

Error: 
```
Error: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at Sh (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:152:147)
    at Rh (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:148:115)
    at Sh (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:151:394)
    at Qj (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:174:181)
    at unstable_runWithPriority (https://example.com/wp-content/plugins/gutenberg/vendor/react.min.755b0ae9.js:24:26)
    at Za (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:73:8)
    at eb (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:170:163)
    at gf (https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:162:317)
    at https://example.com/wp-content/plugins/gutenberg/vendor/react-dom.min.edd8b7a6.js:73:230
    at unstable_runWithPriority (https://example.com/wp-content/plugins/gutenberg/vendor/react.min.755b0ae9.js:24:26)
```

**Comment by syhussaini** [comment]
I am also experiencing the same issue 4795169-zen when hovering over the **Carousel** block. All other blocks work fine.

* Tried to add other blocks and found them working without any issues.
* Disabled the "Gutenberg plugin" to see if that helps. Still getting the same error.

Please see ticket 4795169-zen for the video of the activity.

**Comment by retnonindya** [comment]
Another report here: 34513246-hc

The user informed that no matter _how_ they added the Carousel Block, it keeps giving errors. As for me, it happens when I added Carousel Block by clicking the plus button on the top left and search for Carousel Block there (I am able to add Carousel Block when using `/` method or by clicking the plus button on the content area in the editor.)

Deactivating ALL plugins except Coblocks didn't help either.

The error is similar to @jartes's error with heavy references to Gutenberg. When I checked CLI, I noticed Coblocks is due to update, but I can't update it from CLI since the plugin is symlinked.

I changed the priority from Normal to High as we keep seeing similar reports trickling in lately.

**Comment by danielk-am** [comment]
Also encountered here 34501370-hc.

Switching Image to Carousel block knocked the user out of the Editor.

**Comment by Greatdane** [comment]
Apparently also happens with the Masonry Block; https://github.com/Automattic/wp-calypso/issues/61403

**Comment by danielk-am** [comment]
Encountered a new one here. 4843036-zd-woothemes.

https://user-images.githubusercontent.com/92185881/157389920-941f4c01-1794-4080-ad00-37b726eb14b7.mp4

A screen recording from my test site.

Mouse over the Carousel blocks (while switching from Gallery block) and I got knocked out from the Editor.

**Comment by masperber** [comment]
Another report here: 34551405-hc

I deactivated CoBlocks, and this should prevent the issue from reoccurring with this user.

I'm reporting this mainly so that we can follow up with the user in the event that this issue is closed so that they can turn CoBlocks back on safely.

**Comment by cometgrrl** [comment]
I am not able to reproduce this with Coblocks 2.24.2 (latest version). According to [this PR](https://github.com/godaddy-wordpress/coblocks/pull/2147), the issue was resolved with Coblocks 2.20.0, so if this is reported again, please check the Coblocks plugin version is at least 2.20.0.
